### ISSUE TITLE
Docker will not start in 1192.2.0

### ISSUE THREAD

**Issue by kking124** [coreos/bugs#1647]
# Issue Report #
Docker died last night on my server. It won't restart. 

## Bug ##
```
docker core # systemctl status -l docker
docker.service - Docker Application Container Engine
   Loaded: loaded (/etc/systemd/system/docker.service; disabled; vendor preset: disabled)
   Active: failed (Result: exit-code) since Thu 2016-11-03 15:07:27 UTC; 3min 2s ago
     Docs: http://docs.docker.com
  Process: 1671 ExecStart=/usr/lib/coreos/dockerd daemon --host=fd:// $DOCKER_OPTS $DOCKER_CGROUPS $DOCKER_OPT_BIP $DOCKER_OPT_MTU $DOCKER_OPT_IPMASQ (code=exited, stat
 Main PID: 1671 (code=exited, status=1/FAILURE)

Nov 03 15:07:27 docker systemd[1]: Started Docker Application Container Engine.
Nov 03 15:07:27 docker dockerd[1671]: dockerd: "dockerd" requires 0 arguments.
Nov 03 15:07:27 docker dockerd[1671]: Usage:        dockerd [OPTIONS]
Nov 03 15:07:27 docker systemd[1]: docker.service: Main process exited, code=exited, status=1/FAILURE
Nov 03 15:07:27 docker systemd[1]: docker.service: Unit entered failed state.
Nov 03 15:07:27 docker systemd[1]: docker.service: Failed with result 'exit-code'.
```

### CoreOS Version ###

```
docker core # cat /etc/os-release
NAME=CoreOS
ID=coreos
VERSION=1192.2.0
VERSION_ID=1192.2.0
BUILD_ID=2016-10-21-0026
PRETTY_NAME="CoreOS 1192.2.0 (MoreOS)"
ANSI_COLOR="1;32"
HOME_URL="https://coreos.com/"
BUG_REPORT_URL="https://github.com/coreos/bugs/issues"
```

### Environment ###

Some random cheap optiplex.

XenServer 7.0 Build: 125380c

### Expected Behavior ###

docker to start.

### Actual Behavior ###

docker does not start.

### Reproduction Steps ###

Booted the machine. Docker does not start. Docker will not start when ```systemctl start docker``` command issued either under core or sudo privs. See above error message from journalctl.

**Comment by dm0-** [comment]
From the output you pasted, it looks like you are overriding the system docker service unit with `/etc/systemd/system/docker.service` that has an incorrect command; i.e. `.../dockerd daemon --host=...` is not valid.  It should be `.../docker --host=...`.  You can compare it with `/lib/systemd/system/docker.service`.  Do you still have errors after adjusting the command?

**Comment by kking124** [comment]
It changes the error, but docker still will not start as a service.

```
# systemctl status -l docker
docker.service - Docker Application Container Engine
   Loaded: loaded (/etc/systemd/system/docker.service; disabled; vendor preset: disabled)
   Active: inactive (dead)
     Docs: http://docs.docker.com

Nov 05 15:34:03 docker systemd[1]: Dependency failed for Docker Application Container Engine.
Nov 05 15:34:03 docker systemd[1]: docker.service: Job docker.service/start failed with result 'dependency'.
```

```
# journalctl -xe
Nov 05 15:33:50 docker polkitd[1012]: Acquired the name org.freedesktop.PolicyKit1 on the system bus
Nov 05 15:33:50 docker polkitd[1012]: Registered Authentication Agent for unix-process:1008:4886 (system bus name :1.17 [/usr/bin/pkttyagent --notify-fd 5 --fallback], object path /
Nov 05 15:33:53 docker polkit-agent-helper-1[1031]: pam_unix(polkit-1:auth): authentication failure; logname= uid=500 euid=0 tty= ruser=root rhost=  user=root
Nov 05 15:33:54 docker polkitd[1012]: Operator of unix-process:1008:4886 FAILED to authenticate to gain authorization for action org.freedesktop.systemd1.manage-units for system-bus
Nov 05 15:33:54 docker polkitd[1012]: Unregistered Authentication Agent for unix-process:1008:4886 (system bus name :1.17, object path /org/freedesktop/PolicyKit1/AuthenticationAgen
Nov 05 15:33:59 docker update_engine[817]: I1105 15:33:59.488605   817 update_attempter.cc:481] Updating boot flags...
Nov 05 15:33:59 docker locksmithd[936]: [etcd.service etcd2.service] are inactive
Nov 05 15:33:59 docker locksmithd[936]: Unlocking old locks failed: [etcd.service etcd2.service] are inactive. Retrying in 40s.
Nov 05 15:34:03 docker sudo[1052]:     core : TTY=pts/0 ; PWD=/home/core ; USER=root ; COMMAND=/bin/systemctl start docker
Nov 05 15:34:03 docker sudo[1052]: pam_unix(sudo:session): session opened for user root by core(uid=0)
Nov 05 15:34:03 docker sudo[1052]: pam_systemd(sudo:session): Cannot create session: Already running in a session
Nov 05 15:34:03 docker systemd[1]: Reached target Run Docker containers before main Docker starts up.
-- Subject: Unit early-docker.target has finished start-up
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- Unit early-docker.target has finished starting up.
--
-- The start-up result is done.
Nov 05 15:34:03 docker systemd[1]: docker.socket: Failed to listen on sockets: No such file or directory
Nov 05 15:34:03 docker systemd[1]: Failed to listen on Docker Socket for the API.
-- Subject: Unit docker.socket has failed
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- Unit docker.socket has failed.
--
-- The result is failed.
Nov 05 15:34:03 docker systemd[1]: Dependency failed for Docker Application Container Engine.
-- Subject: Unit docker.service has failed
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- Unit docker.service has failed.
--
-- The result is dependency.
Nov 05 15:34:03 docker systemd[1]: docker.service: Job docker.service/start failed with result 'dependency'.
Nov 05 15:34:03 docker sudo[1052]: pam_unix(sudo:session): session closed for user root
Nov 05 15:34:28 docker sudo[1057]:     core : TTY=pts/0 ; PWD=/home/core ; USER=root ; COMMAND=/bin/su
Nov 05 15:34:28 docker sudo[1057]: pam_unix(sudo:session): session opened for user root by core(uid=0)
Nov 05 15:34:28 docker sudo[1057]: pam_systemd(sudo:session): Cannot create session: Already running in a session
Nov 05 15:34:29 docker su[1058]: Successful su for root by root
Nov 05 15:34:29 docker su[1058]: + /dev/pts/0 root:root
Nov 05 15:34:29 docker su[1058]: pam_unix(su:session): session opened for user root by core(uid=0)
Nov 05 15:34:29 docker su[1058]: pam_systemd(su:session): Cannot create session: Already running in a session
Nov 05 15:34:39 docker locksmithd[936]: [etcd.service etcd2.service] are inactive
Nov 05 15:34:39 docker locksmithd[936]: Unlocking old locks failed: [etcd.service etcd2.service] are inactive. Retrying in 1m20s.
Nov 05 15:35:59 docker locksmithd[936]: [etcd.service etcd2.service] are inactive
Nov 05 15:35:59 docker locksmithd[936]: Unlocking old locks failed: [etcd.service etcd2.service] are inactive. Retrying in 2m40s.
Nov 05 15:38:39 docker locksmithd[936]: [etcd.service etcd2.service] are inactive
Nov 05 15:38:39 docker locksmithd[936]: Unlocking old locks failed: [etcd.service etcd2.service] are inactive. Retrying in 5m0s.
```

Please keep in mind that this docker instance was running fine and then crashed in the middle of operations without any changes to the configuration instigated on my part. I assume it's related to an automatic update that occurred on the system, but I have not made changes knowingly to this coreos VM's configuration since it was deployed on May 30, 2016.

**Comment by crawford** [comment]
You are using a custom `docker.service` which we can't guarantee will work as we make changes to the system. We recommend using drop-ins to set environment variables which control the behavior of Docker. Can you show the contents of `docker.service` so I can see what you are trying to do? (`systemctl cat docker`)

**Comment by dm0-** [comment]
This looks like the above linked issue, which seems to be fixed in docker/machine#3870.  You can reopen this if that is not the case.
